confidential-data-hub/secret/tests/secret_cli.rs

```diff
@@ -0,0 +1,187 @@
+    let cert_path: &str = &format!("{}{}", base_dir, "PrivateKmsCA_kst-bjj6____.pem");
+    let kms_instance_id: &str = "kst-bjj6****";
+    let client_key_file_path: &str = &format!("{}{}", base_dir, "clientKey_KAAP.2bc4____.json");
+
+    let dir = tempdir().expect("create temdir fail");
+    let dir_path = dir.path();
+
+    let original_secret_file_path = dir_path.join("plaintext").to_str().unwrap().to_owned();
+    let sealed_secret_file_path = dir_path
+        .join("sealed_secret.json")
+        .to_owned()
+        .to_str()
+        .unwrap()
+        .to_owned();
+    let unsealed_secret_file_path = format!("{}.unsealed", &sealed_secret_file_path);
+
+    // create random secret
+    let secret = create_random_secret();
+    File::create(&original_secret_file_path)
+        .expect("create 'secret_file' fail")
+        .write_all(&secret)
+        .expect("write 'secret_file' fail");
+
+    // seal secret with 'secret_cli'
+    let seal_secret_output = Command::cargo_bin("secret_cli")
+        .expect("init 'cargo_bin' fail")
+        .arg("seal")
+        .args(["--file-path", &original_secret_file_path])
+        .arg("envelope")
+        .args(["--key-id", key_id])
+        .arg("ali")
+        .args(["--password-file-path", client_key_password_file_path])
+        .args(["--kms-instance-id", kms_instance_id])
+        .args(["--client-key-file-path", client_key_file_path])
+        .args(["--cert-path", cert_path])
+        .output()
+        .expect("Failed to execute seal secret");
+
+    File::create(&sealed_secret_file_path)
+        .expect("create 'sealed secret_file' fail")
+        .write_all(&seal_secret_output.stdout)
+        .expect("write 'sealed_secret_file' fail");
+
+    assert!(seal_secret_output.status.success());
+
+    // unseal secret with 'secret_cli'
+    let unseal_secret_output = Command::cargo_bin("secret_cli")
+        .expect("init 'cargo_bin' fail")
+        .arg("unseal")
+        .args(["--file-path", &sealed_secret_file_path])
+        .args(["--key-path", &base_dir])
+        .output()
+        .expect("Failed to execute unseal secret");
+
+    assert!(unseal_secret_output.status.success());
+
+    // compare original secret and unsealed secret
+    let mut original_secret_file =
+        File::open(&original_secret_file_path).expect("open original_secret_file fail");
+    let mut original_secret = Vec::new();
+    original_secret_file
+        .read_to_end(&mut original_secret)
+        .expect("read original_secret_file fail");
+
+    let mut unsealed_secret_file =
+        File::open(&unsealed_secret_file_path).expect("open unsealed_secret_file fail");
+    let mut unsealed_secret = Vec::new();
+    unsealed_secret_file
+        .read_to_end(&mut unsealed_secret)
+        .expect("read unsealed_secret_file fail");
+
+    assert_eq!(original_secret, unsealed_secret);
+}
+
+#[test]
+#[ignore]
+#[cfg(feature = "ehsm")]
+fn test_cli_envlope_secret_lifetime_ehsm() {
```

Comment:
There is a lot of repetition in this function (with the one above. Is there a way to make some helper functions that can be shared?